### ISSUE TITLE
Unrecognized literal string suffixes

### ISSUE THREAD

**Issue by j-cortial** [jeff-hykin/better-cpp-syntax#610]
Checklist

- [X] This problem exists even with the setting `"C_Cpp.enhancedColorization": "Disabled"`

## The code with a problem is:
```cpp
#include <string>
#include <string_view>

using namespace std::literals::string_view_literals;
using namespace std::literals::string_literals;

const auto view = "Hello"sv;
const auto owned = "Hello"s;
```

## It looks like:
![string_suffix](https://user-images.githubusercontent.com/108632234/198851172-ef6464e7-c424-40d1-88a9-48448d49b75e.png)

## It should look like:
The suffixes "s" and "sv" should be highlighted.

## Reference
https://en.cppreference.com/w/cpp/language/user_literal

**Comment by jeff-hykin** [comment]
User literals should be covered (the ref link) but I was unaware of std literals till now.

https://en.cppreference.com/w/cpp/string/basic_string_view/operator%22%22sv

https://en.cppreference.com/w/cpp/string/basic_string/operator%22%22s

https://en.cppreference.com/w/cpp/numeric/complex/operator%22%22i


https://en.cppreference.com/w/cpp/symbol_index/literals

**Comment by j-cortial** [comment]
Numeric user-defined literal suffixes are already covered, but not suffixes for strings and characters.
I am currently working on a PR to fix that.

All literal suffixes are "user-defined", even the ones provided by the standard library. However, the suffixes that do not start with an underscore are reserved for use by the standard library (such as [std::string operator""s](https://en.cppreference.com/w/cpp/string/basic_string/operator%22%22s)), and they should not be defined in a program.

By the way, I think the tag used for numeric literal suffixes (`keyword.other.unit.user-defined`) could be improved, since there no intrinsic notion of unit (except in some special cases, for `std::chrono` for instance) for literals in general.
I would rather suggest `keyword.other.suffix.user-defined.numeric` (and `keyword.other.suffix.user-defined.reserved.numeric` for those not starting with an underscore) for instance.
The same remark applies to language-defined suffixes such as `keyword.other.unit.suffix.floating-point` and friends, since they specify a type, not a unit.

**Comment by jeff-hykin** [comment]
Sadly the `keyword.other.unit` is not done because it is accurate, but because `keyword.other.unit` is used by other languages and themes, while inventing a new scope means absolutely no existing theme is going to highlight it.

Ex: CSS highlighting
<img width="700" alt="Screen Shot 2022-12-10 at 6 24 04 PM" src="https://user-images.githubusercontent.com/17692058/206880709-d4e7ffa0-3290-4a81-ac52-87d8eeeaeb28.png">

However we can do both; `"keyword.other.unit keyword.other.suffix.user-defined.numeric"` will let it default to the `keyword.other.unit` color, but be overridden for super-customized themes that actually know about the existance of `keyword.other.suffix.user-defined.numeric`

**Comment by j-cortial** [comment]
My proposal in the PR: https://github.com/jeff-hykin/better-cpp-syntax/pull/622

**Comment by jeff-hykin** [comment]
published with `v1.17.3`! thanks for handling this
